cfgov/unprocessed/js/modules/multi-select.js

```diff
@@ -0,0 +1,494 @@
+'use strict';
+
+function init() {
+  var _ = function ( elem, opts ) {
+    var self = this;
+
+    this.elem = $( elem );
+    opts = opts || {};
+
+    configure.call( this, {
+      minChars:  3,
+      maxItems:  10,
+      autoFirst: false,
+      filter: _.FILTER_CONTAINS
+    }, opts );
+
+    this.index = -1;
+    this.name = this.elem.name;
+    this.skipBlur = false;
+    this.optionData = [];
+    this.selections = [];
+    this.options = this.elem.options || [];
+    this.placeholder;
+
+    _.all.push( this );
+  };
+
+  _.prototype = {
+    set options( options ) {
+      this.placeholder = this.elem.getAttribute( 'placeholder' );
+      this.sanitizeList( options );
+
+      if ( this.optionData.length > 0 ) {
+        this.populateMarkup();
+        this.bindEvents();
+      }
+    },
+
+    sanitizeList: function( list ) {
+      var self = this;
+      var l = list.length;
+
+      for ( var i = 0; i < l; i++ ) {
+        if ( list[i].dataset.cflabel === undefined ) {
+          this.optionData.push( {
+            value: list[i].value,
+            text:  list[i].text
+          } );
+        } else {
+          if ( !self.placeholder ) {
+            self.placeholder = list[i].text;
+          }
+        }
+      }
+
+      this.filtered = this.optionData;
+    },
+
+    populateMarkup: function() {
+      // Create necessary elements
+      var self = this;
+
+      // Add a container for our markup and hide the default select elem
+      this.container = $.create( 'div', {
+        className: 'cf-multi-select',
+        around:    self.elem
+      } );
+
+      // Create all our markup but wait to manipulate the DOM just once
+      this.header = $.create( 'header', {
+        className: 'cf-multi-select_header'
+      } );
+
+      this.choices = $.create( 'ul', {
+        className: 'list__unstyled cf-multi-select_choices',
+        inside:    self.container
+      } );
+
+      this.search = $.create( 'input', {
+        className:   'cf-multi-select_search',
+        type:        'text',
+        placeholder: self.placeholder || 'Choose up to five',
+        inside:      self.header
+      } );
+
+      this.fieldset = $.create( 'fieldset', {
+        className:  'cf-multi-select_fieldset',
+        visibility: 'hidden'
+      } );
+
+      this.list = $.create( 'ul', {
+        className: 'list__unstyled cf-multi-select_options',
+        inside:    self.fieldset
+      } );
```

Comment:
`innerHTML` wouldn't work because I'm manipulating each of these later on, but templating might work. cc @sebworks any thoughts on what's best in this situation? Did we ever settle on one engine?